todo-app/src/App.js

```diff
@@ -1,13 +1,83 @@
-import React, { Component } from 'react';
+//jshint esnext: true
+
+import React from 'react';
+
+import TodoList from './Components/todoList';
+import AddTodoItem from './Components/AddTodoItem';
+import './App.css';
+
+
+export default class App extends React.Component {
+
+  
+  state= {
+    todos: [],
+    isEditing: null
+  }
+
+
+  handleToggleEditing = (id) => {
+    this.setState({ isEditing: id || null })
+  }
+
+  handleAddTodo = (text, deadline) => {
+    
+    const newItem = {
+        id: this.state.todos.length + 1 || 1,
+        text,
+        deadline,
+        done: false
+      }
+      const oldTodos = this.state.todos || [];
+      
+      this.setState({ todos: [...oldTodos, newItem]})
+    }
+
+  handleToggleDone = (id) => {
+    const newTodos = [...this.state.todos].map(item => {
+      if (item.id === id){
+        item.done = !item.done; 
+      }
+      return item;
+    });
+    this.setState({todos: newTodos});
+  }
+
+  handleRemove = (id) => {
+    const newTodos = [...this.state.todos].filter(item => item.id !== id);
```

Comment:
Here it's okay to `filter` upon `this.state.todos` directly, as filter doesn't mutate any values, it just returns a new array.